staging/src/k8s.io/client-go/rest/config.go

```diff
@@ -128,6 +129,12 @@ type Config struct {
 	// Dial specifies the dial function for creating unencrypted TCP connections.
 	Dial func(ctx context.Context, network, address string) (net.Conn, error)
 
+	// Proxier is the the proxier to be used for all requests made by this
+	// transport.
+	//
+	// socks5 proxying does not currently support spdy streaming endpoints.
+	Proxier func(*http.Request) (*url.URL, error)
```

Comment:
unless we have a good reason, suggest naming the same as http.Transport (`Proxy`)

Would it make sense to define similar behavior to http.Transport around returned nil values, e.g.:
```
If Proxy is nil, http.ProxyFromEnvironment is used.
If Proxy returns a nil *URL, no proxy is used.
```

Reply:
Done.